src/wp-includes/theme.php

```diff
@@ -4352,6 +4352,11 @@ function create_initial_theme_features() {
  * @return bool Whether the active theme is a block-based theme or not.
  */
 function wp_is_block_theme() {
+	if ( ! did_action( 'setup_theme' ) ) {
```

Comment:
After more testing, I realized that I made an incorrect observation about the return value of `wp_is_block_theme()` before themes are set up. If not for the early return we have here, it will return the correct value regardless of when it's called. Given that, we really only need to protect against this being called before `register_theme_directory()` has been called here:
https://github.com/WordPress/wordpress-develop/blob/13fe8be4359222b7a53d4348dbd5fa81abe7080d/src/wp-settings.php#L519

I think this is a much safer approach.

```suggestion
	if ( empty( $GLOBALS['wp_theme_directories'] ) ) {
```